R/stringify_expression.R

```diff
@@ -1,66 +1,32 @@
 ## input is quote()'ed expression
 ## reprex() takes care of that or, to use directly:
 ## q <- quote({a + b})
-stringify_expression <- function(q) {
-  src <- attr(q, "srcref")
 
-  ## if input is, eg 1:5, there is no srcref
-  if (is.null(src)) {
-    return(deparse(q))
-  }
-
-  ## https://journal.r-project.org/archive/2010-2/RJournal_2010-2_Murdoch.pdf
-  ## Each element is a vector of 6 integers: (first line, first byte, last line,
-  ## last byte, first character, last character)
-  ## ^^ clearly not whole story because I see vectors of length 8
-  ## but seems to be true description of first 6 elements
-  coord <- do.call(rbind, lapply(src, `[`, c(1, 5, 3, 6)))
-  colnames(coord) <- c("first_line", "first_character",
-                       "last_line", "last_character")
-  n_statements <- nrow(coord)
-  ## each row holds coordinates of source for one statement
-
-  ## all statements seem to contain such a reference to same lines of source
-  lines <-  attr(src[[1]], "srcfile")$lines
+#' @importFrom utils head tail getSrcref
+stringify_expression <- function(x, fun = as.character(sys.call()[[1L]])) {
```

Comment:
Is `fun = as.character(sys.call()[[1L]])` leftover from some intermediate investigation?

Reply:
Oh yeah, sorry will remove